presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/db/ResourceGroupsDao.java

```diff
@@ -93,9 +93,11 @@
 
     @SqlQuery("SELECT resource_group_id\n" +
             "FROM exact_match_source_selectors\n" +
-            "WHERE environment = :environment\n" +
-            "  AND source = :source\n" +
-            "  AND query_type = :query_type")
+            "WHERE source = :source\n" +
+            "  AND (environment = :environment OR environment IS NULL)\n" +
+            "  AND (query_type = :query_type OR query_type IS NULL)\n" +
+            "ORDER BY environment, query_type DESC\n" + // puts more specific selectors first
```

Comment:
MySQL does nulls first for ASC, nulls last for DESC. However, the reader needs to know about that behavior and that this is intended (which is more confusing since it's opposite for `environment` and `query_type`). Add Javadoc to the method explaining the intended return order.

Reply:
Is this behavior specific to MySQL? Is only MySQL supported here?

Reply:
Today, this plugin only supports H2 and MySQL, but we should make this code work for other databases (suppose we ad PostgreSQL later). We decided on `ORDER BY x IS NULL` since that's portable and we only care about null ordering (it either matches exact or null).

Reply:
> We decided on `ORDER BY x IS NULL` since that's portable 

Appreciated :)